### ISSUE TITLE
[Meta] Mac M1 Support 

### ISSUE THREAD

**Issue by ph** [elastic/beats#26396]
Before starting the following we need to answer how we are supporting m1 mac, there are two way we can do this.

1. Different binary and packages.
2. A universal binary, with a quick search this might be possible. https://dev.to/thewraven/universal-macos-binaries-with-go-1-16-3mm3

----

1. Allow beats to compile for m1 machine, we have recently updated to Go 1.16.x that was blocking this effort.
2. Investigate if we need a special crossbuild docker images to build for m1. (https://github.com/elastic/golang-crossbuild)
3. Add support for Fleet-Server M1.
4. Enable Jenkins Pipeline for m1 workers 
5. Investigate what kind of test we can run, Unit, System Test, etc.
6. Update release manager with the m1 artifact.

**Comment by ChefAustin** [comment]
As a consumer of Beats products, I wanted to weigh-in and request that Universal2 packages be prioritized (over different packages being made for both ARM64 and Intel architectures).

Offering Universal2 packages provides Elastic's customers with a much simpler means of deployment (insofar that deployment logic doe not need to take into account architecture given that a Universal2 package will "just work" on any device it would be deployed to).

While I understand that Universal2 packages are a bit more complex to build, the amount of headache that would be mitigated for your customers is YUGE. My organization is having a similar discussion with a lot of our vendors about this topic and -- from a customer perspective -- there is no question that Universal2 is the way to go.

My team currently manages a fleet of ~25k macOS devices and as we move hardware deployments toward Apple's M1 SoC, it would be of great benefit to have Universal2 Beats packages that will gracefully install on whatever hardware they're pushed to. Without a Universal2 package, we would then have to write additional logic to ensure that non-M1 machines get pushed the Intel package and M1 machines get the ARM64 package. While this might not seem like a big deal, it would be -- in my eyes -- unnecessary work. I say it would be unnecessary work because this is the exact reason why Apple has provided macOS application development companies the Universal2 binary: To aid in the transition between Intel and Apple SoC. That being said, for a development company to not lean-in and build Universal2 binaries equates to the development company not caring enough about their customers (and, in this case, paying customers) and merely "passing the buck" down to them to figure out.

**tl;dr - I can't stress enough how much our org would prefer Universal2 packages over arch-specific packages for Beats products.**

**Comment by jlind23** [comment]
Depending on: https://github.com/elastic/observability-robots/issues/871
